src/app-gocardless/services/gocardless-service.js

```diff
@@ -244,6 +242,69 @@ export const goCardlessService = {
     };
   },
 
+  /**
+   *
+   * @param requisitionId
+   * @param accountId
+   * @param startDate
+   * @param endDate
+   * @throws {AccountNotLinedToRequisition} Will throw an error if requisition not includes provided account id
+   * @throws {RequisitionNotLinked} Will throw an error if requisition is not in Linked
+   * @throws {InvalidInputDataError}
+   * @throws {InvalidGoCardlessTokenError}
+   * @throws {AccessDeniedError}
+   * @throws {NotFoundError}
+   * @throws {ResourceSuspended}
+   * @throws {RateLimitError}
+   * @throws {UnknownError}
+   * @throws {ServiceError}
+   * @returns {Promise<{institutionId: string, transactions: {booked: Array<import('../gocardless-node.types.js').Transaction>, pending: Array<import('../gocardless-node.types.js').Transaction>, all: Array<import('../gocardless.types.js').TransactionWithBookedStatus>}}>}
+   */
+  getNormalizedTransactions: async (
+    requisitionId,
+    accountId,
+    startDate,
+    endDate,
+  ) => {
+    const { institution_id, accounts: accountIds } =
+      await goCardlessService.getLinkedRequisition(requisitionId);
+
+    if (!accountIds.includes(accountId)) {
+      throw new AccountNotLinedToRequisition(accountId, requisitionId);
+    }
+
+    const transactions = await goCardlessService.getTransactions({
+      institutionId: institution_id,
+      accountId,
+      startDate,
+      endDate,
+    });
+
+    const bank = BankFactory(institution_id);
+    const sortedBookedTransactions = bank.sortTransactions(
+      transactions.transactions?.booked,
+    );
+    const sortedPendingTransactions = bank.sortTransactions(
+      transactions.transactions?.pending,
+    );
+    const allTransactions = sortedBookedTransactions.map((t) => {
+      return { ...t, booked: true };
+    });
+    sortedPendingTransactions.forEach((t) =>
+      allTransactions.push({ ...t, booked: false }),
+    );
+    const sortedAllTransactions = bank.sortTransactions(allTransactions);
+
```

Comment:
Can we refactor this out into a shared method to avoid the duplication with the other function?  That way there is less chance of divergence.